src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+import os
+import time
+from typing import Any, Callable, Union
+from uuid import uuid4
+from ..client import Vellum
+from ..types import (
+    ChatMessage,
+    NamedTestCaseJsonVariableValueRequest,
+    NamedTestCaseNumberVariableValueRequest,
+    NamedTestCaseStringVariableValueRequest,
+    SearchResult,
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    VellumError,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(self, id: str) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self.test_suite_run_state = TestSuiteRunState.QUEUED
+        self.test_suite_run_executions: list[TestSuiteRunExecution] | None = None
+
+    def _refresh_test_suite_run_state(self):
+        test_suite_run = self.client.test_suite_runs.retrieve(self.id)
+        self.test_suite_run_state = test_suite_run.state
+
+    def _ensure_metric_results_are_loaded(self):
+        if self.test_suite_run_executions is not None:
+            return
+
+        self._refresh_test_suite_run_state()
+
+        while self.test_suite_run_state in {TestSuiteRunState.QUEUED, TestSuiteRunState.RUNNING}:
+            time.sleep(1)
+            self._refresh_test_suite_run_state()
```

Comment:
[Aesthetic Opinion] another way to write this that avoids two references to `_refresh_test_suite_run_state`
```suggestion
        while True:
            self._refresh_test_suite_run_state()
            if self.test_suite_run_state not in {TestSuiteRunState.QUEUED, TestSuiteRunState.RUNNING}:
                break
            time.sleep(1)
```

Reply:
while True's scare me but I'll switch it